dnf/module/repo_module_version.py

```diff
@@ -0,0 +1,205 @@
+                    self.base.install(nevr, reponame=self.repo.id, forms=hawkey.FORM_NEVR,
+                                      strict=strict)
+                    self.base._goal.group_members.add(nevra_object.name)
+                    result = True
+
+        return result
+
+    def upgrade(self, profiles):
+        installed = self.base.sack.query().installed().run()
+        installed_nevras = [str(pkg) for pkg in installed]
+        query_to_return = None
+
+        for profile in profiles:
+            if profile not in self.profiles:
+                raise NoProfileException("{}/{}".format(self.full_version, profile))
+
+            for nevra_object in self.profile_nevra_objects(profile):
+                nevr = self.nevra_object_to_nevr_str(nevra_object)
+                nevra = "{}.{}".format(nevr, nevra_object.arch)
+
+                if nevra not in installed_nevras:
+                    self.base.install(nevr, reponame=self.repo.id, forms=hawkey.FORM_NEVR,
+                                      strict=self.base.conf.strict)
+                else:
+                    # TODO: verify that filter(nevra) really works correctly
+                    # (possibly breaks multilib)
+                    query = self.base.sack.query().filter(nevra=nevra)
+                    if query_to_return is None:
+                        query_to_return = query
+                    else:
+                        query_to_return = query_to_return.union(query)
+
+        self.base._module_persistor.set_data(self.repo_module, stream=self.stream,
+                                             version=self.version)
+
+        return query_to_return
+
+    def remove(self, profiles):
+        for profile in profiles:
+            if profile not in self.profiles:
+                raise NoProfileException("{}/{}".format(self.full_version, profile))
+
+            for nevra_object in self.profile_nevra_objects(profile):
+                nevr = self.nevra_object_to_nevr_str(nevra_object)
+                remove_query = dnf.subject.Subject(nevr) \
+                    .get_best_query(self.base.sack, forms=hawkey.FORM_NEVR)
+
+                if not remove_query \
+                        or self.base._yumdb.get_package(remove_query[0]).reason == 'user':
+                    continue
+
+                self.base._remove_if_unneeded(remove_query)
+
+        conf = self.repo_module.conf
+        version = conf.version._get()
+        profiles = [x for x in list(conf.profiles._get()) if x not in profiles]
+
+        if len(list(conf.profiles._get())) == 0:
+            conf.version._set(-1)
+
+        self.base._module_persistor.set_data(self.repo_module, stream=self.stream, version=version,
+                                             profiles=sorted(set(profiles)))
+
+    def nevra(self):
+        result = self.artifacts()
+        # HACK: remove epoch to make filter(nevra=...) work
+        result = [i.replace("-0:", "-") for i in result]
```

Comment:
Why this is needed here. Even nevra or nevra_strict filters can handle both cases.

Reply:
Old, unnecessary code, I already forgot about. Removed.